src/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,494 @@
+				timeLong = ((aveRtt + SLEEP_TIME_LONG) < RTT_MAX ? (SLEEP_TIME_LONG + aveRtt) : RTT_MAX)/3;
+				String sleepInj;
+				
+			while(index<SLEEP_INJECTION.length) {
+				if(isStop()) {
+					if (LOG.isDebugEnabled()) { LOG.debug("Stopping the scan due to a user request"); }
+					return;
+				}
+				if (LOG.isDebugEnabled()) {
+					LOG.debug("\nTrying  with the value: "+SLEEP_INJECTION[index][0]);
+				}
+				try {
+					sleepInj = SLEEP_INJECTION[index][0].replaceFirst(INSERT_TIME_SHORT, Integer.toString(timeShort));
+					HttpMessage injMsg =  getNewMsg();
+					setParameter(injMsg, param, sleepInj);
+					sendAndReceive(injMsg, false);
+					if(injMsg.getTimeElapsedMillis()-aveRtt >= timeShort-DELTA_TIME) {
+						// try for a longer time to exclude transmission delays
+						if(isStop()) {
+							if (LOG.isDebugEnabled()) { LOG.debug("Stopping the scan due to a user request"); }
+							return;
+						}
+						if (LOG.isDebugEnabled()) {
+							LOG.debug("\nTrying for a longer time with the value: "+SLEEP_INJECTION[index][1]);
+						}
+						sleepInj = SLEEP_INJECTION[index][1].replaceFirst(INSERT_TIME_LONG, Integer.toString(timeLong));
+						injMsg =  getNewMsg();
+						setParameter(injMsg, param, sleepInj);
+						sendAndReceive(injMsg, false);
+						if(injMsg.getTimeElapsedMillis()-aveRtt >= timeLong-DELTA_TIME) {
+							bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH, getName(), getDescription(), null,
+									param, SLEEP_INJECTION[index][1], getExtraInfo(SLEEP_ATTACK), getSolution(), injMsg);
+							isBingo = true;
+							break;
+						}
+					}
+					index++;
+				} catch (SocketTimeoutException ex) {
+					if(SLEEP_TIME_SHORT!=3000) {
+						timeShort = (aveRtt + 3000)/3;
+						timeLong = (aveRtt + 5000)/3;
+						if (LOG.isDebugEnabled()) {
+							LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() +" when "+
+									IO_EX_LOG+"due to a socket timeout, trying with the lowest interval("+timeLong+")");
+						}
+					}
+					else { break; }
+				} catch (IOException ex) {
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() +" when "+IO_EX_LOG);
+					}
+					return;
+				}
+			}
+		}
+		// json query injection
+		if(doJsonScan && isJsonPayload) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("\nStarting with the json query injection package of attack rules:");
+			}
+			for(String[] jpv : JSON_INJECTION) {
+				try {
+					if(isStop()) {
+						if (LOG.isDebugEnabled()) { LOG.debug("Stopping the scan due to a user request"); }
+						return;
+					}
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("\nTrying with the value: "+jpv);
+					}
+					String valueInj = getParamJsonString(param, jpv);
+					HttpMessage injMsg = getNewMsg();
+					injMsg.getRequestHeader().setHeader(HttpRequestHeader.CONTENT_TYPE, "application/json");
+					injMsg.getRequestHeader().setMethod(HttpRequestHeader.POST);

```

Comment:
Should this only be done on requests that are POST to start with?

Reply:
Yes, in fact this line is useless. I removed it.